src/MOI/MOI_wrapper.jl

```diff
@@ -2924,33 +2944,81 @@ function MOI.optimize!(model::Optimizer)
     model.termination_status = _cache_termination_status(model)
     model.primal_status = _cache_primal_status(model)
     model.dual_status = _cache_dual_status(model)
-    # TODO: add @checked here - must review statuses
-    if model.has_nlp_constraints
-        Lib.XPRSgetnlpsol(
+
+    # XPRSgetnlpsol and XPRSgetmippsol are deprecated as of version 44.01
+    # XPRSgetsolution is also recommended instead of XPRSgetlpsol also after solving LPs
+    if version >= v"44.01"
```

Comment:
Were you very careful selecting this and the other version?

Reply:
Deprecations are given with version in `xpress.h`. For example
```
XPRS_EXPORT
xprs_deprecated("since 44.00")
int XPRS_CC XPRSgetnlpsol(XPRSprob prob, double x[], double slack[], double duals[], double djs[]);
```
As for the introduction of `XPRSoptimize`, I have checked that it is present in v41 (Xpress 9.0) and not in v40 (Xpress 8.14.x).